sys/include/auto_init_macros.h

```diff
@@ -0,0 +1,425 @@
+    AUTO_INIT_PRIO_AFTER(AUTO_INIT_PRIO_HIGHEST, CONFIG_AUTO_INIT_PRIO_DHCPV6_CLIENT)
+#ifndef CONFIG_AUTO_INIT_PRIO_DHCPV6_RELAY
+#define CONFIG_AUTO_INIT_PRIO_DHCPV6_RELAY                  44
+#endif
+#define AUTO_INIT_PRIO_DHCPV6_RELAY                         \
+    AUTO_INIT_PRIO_AFTER(AUTO_INIT_PRIO_HIGHEST, CONFIG_AUTO_INIT_PRIO_DHCPV6_RELAY)
+#ifndef CONFIG_AUTO_INIT_PRIO_GNRC_DHCPV6_CLIENT_SIMPLE_PD
+#define CONFIG_AUTO_INIT_PRIO_GNRC_DHCPV6_CLIENT_SIMPLE_PD  45
+#endif
+#define AUTO_INIT_PRIO_GNRC_DHCPV6_CLIENT_SIMPLE_PD         \
+    AUTO_INIT_PRIO_AFTER(AUTO_INIT_PRIO_HIGHEST, CONFIG_AUTO_INIT_PRIO_GNRC_DHCPV6_CLIENT_SIMPLE_PD)
+#ifndef CONFIG_AUTO_INIT_PRIO_GNRC_IPV6_AUTO_SUBNETS
+#define CONFIG_AUTO_INIT_PRIO_GNRC_IPV6_AUTO_SUBNETS        46
+#endif
+#define AUTO_INIT_PRIO_GNRC_IPV6_AUTO_SUBNETS               \
+    AUTO_INIT_PRIO_AFTER(AUTO_INIT_PRIO_HIGHEST, CONFIG_AUTO_INIT_PRIO_GNRC_IPV6_AUTO_SUBNETS)
+#ifndef CONFIG_AUTO_INIT_PRIO_MULTIMEDIA
+#define CONFIG_AUTO_INIT_PRIO_MULTIMEDIA                    47
+#endif
+#define AUTO_INIT_PRIO_MULTIMEDIA                           \
+    AUTO_INIT_PRIO_AFTER(AUTO_INIT_PRIO_HIGHEST, CONFIG_AUTO_INIT_PRIO_MULTIMEDIA)
+#ifndef CONFIG_AUTO_INIT_PRIO_SCREEN
+#define CONFIG_AUTO_INIT_PRIO_SCREEN                        48
+#endif
+#define AUTO_INIT_PRIO_SCREEN                               \
+    AUTO_INIT_PRIO_AFTER(AUTO_INIT_PRIO_HIGHEST, CONFIG_AUTO_INIT_PRIO_SCREEN)
+#ifndef CONFIG_AUTO_INIT_PRIO_BENCHMARK_UDP
+#define CONFIG_AUTO_INIT_PRIO_BENCHMARK_UDP                 49
+#endif
+#define AUTO_INIT_PRIO_BENCHMARK_UDP                        \
+    AUTO_INIT_PRIO_AFTER(AUTO_INIT_PRIO_HIGHEST, CONFIG_AUTO_INIT_PRIO_BENCHMARK_UDP)
+/** @} */
+
+#if ENABLE_DEBUG
+#define AUTO_INIT_MODULE(module, prio, function)                                        \
+    static inline void CONCAT(CONCAT(_debug_, prio), function)(void) {                  \
+        DEBUG("auto_init: %s\n", __func__);                                             \
+        function();                                                                     \
+    }                                                                                   \
+    XFA_USE_CONST(auto_init_fn, auto_init_xfa);                                         \
+    XFA_CONST(auto_init_xfa, prio)                                                      \
+    auto_init_fn CONCAT(CONCAT(auto_init_xfa, prio), module) =                          \
+        (auto_init_fn)CONCAT(CONCAT(_debug_, prio), function);
+#else
+/**
+ * @brief   Add a module to the auto-initialization array
+ *
+ * @param   module      Module name
+ * @param   prio        Priority level
+ * @param   function    Function to be called on initialization
+ *
+ * @experimental
+ */
+#define AUTO_INIT_MODULE(module, prio, function)                                        \
+    XFA_USE_CONST(auto_init_fn, auto_init_xfa);                                         \
```

Comment:
I think XFA_USE_X is not needed when just using XFA_X. Could you check?

Reply:
yep, you are right.